contracts/ExampleNFT-v2.cdc

```diff
@@ -0,0 +1,425 @@
+
+    pub event Mint(id: UInt64, type: Type,
+                   displayView: MetadataViews.Display?,
+                   serialView: MetadataViews.Serial?)
+
+    pub event Destroy(id: UInt64, type: Type,
+                      displayView: MetadataViews.Display?,
+                      serialView: MetadataViews.Serial?)
+
+    /// Path where the minter should be stored
+    /// The standard paths for the collection are stored in the collection resource type
+    pub let MinterStoragePath: StoragePath
+
+    /// We choose the name NFT here, but this type can have any name now
+    /// because the interface does not require it to have a specific name any more
+    pub resource NFT: NonFungibleToken.NFT, MetadataViews.Resolver {
+
+        /// The ID of the NFT
+        /// Could be a project specific ID, or the UUID
+        /// Here we choose the UUID
+        pub let id: UInt64
+
+        /// From the Display metadata view
+        pub let name: String
+        pub let description: String
+        pub let thumbnail: String
+
+        /// For the Royalties metadata view
+        access(self) let royalties: [MetadataViews.Royalty]
+
+        /// Generic dictionary of traits the NFT has
+        access(self) let metadata: {String: AnyStruct}
+    
+        init(
+            name: String,
+            description: String,
+            thumbnail: String,
+            royalties: [MetadataViews.Royalty],
+            metadata: {String: AnyStruct},
+        ) {
+            self.id = self.uuid
+            self.name = name
+            self.description = description
+            self.thumbnail = thumbnail
+            self.royalties = royalties
+            self.metadata = metadata
+        }
+    
+        pub fun getViews(): [Type] {
+            return [
+                Type<MetadataViews.Display>(),
+                Type<MetadataViews.Royalties>(),
+                Type<MetadataViews.Editions>(),
+                Type<MetadataViews.ExternalURL>(),
+                Type<MetadataViews.NFTCollectionData>(),
+                Type<MetadataViews.NFTCollectionDisplay>(),
+                Type<MetadataViews.Serial>(),
+                Type<MetadataViews.Traits>()
+            ]
+        }
+
+        pub fun resolveView(_ view: Type): AnyStruct? {
+            switch view {
+                case Type<MetadataViews.Display>():
+                    return MetadataViews.Display(
+                        name: self.name,
+                        description: self.description,
+                        thumbnail: MetadataViews.HTTPFile(
+                            url: self.thumbnail
+                        )
+                    )
+                case Type<MetadataViews.Editions>():
+                    // There is no max number of NFTs that can be minted from this contract
+                    // so the max edition field value is set to nil
+                    let editionInfo = MetadataViews.Edition(name: "Example NFT Edition", number: self.id, max: nil)
```

Comment:
Can we use something else then id here? It is misleading to have uuid as edition

Reply:
What would we use? We don't really have anything else in this contract, right? Should we just remove the edition view?

Reply:
Either just have a counter that runs if they are all with the same name, or just leave it out TBH.